### ISSUE TITLE
Eclipse m2e generated-sources not added when using JAXB specific version

### ISSUE THREAD

**Issue by DeezCashews** [highsource/maven-jaxb2-plugin#94]
Background: I have a project that requires use of `<jaxb:globalBindings optionalProperty="primitive">`.  There is a bug in XJC 2.2 that breaks with NPE when using this.  Using `maven-jaxb2-plugin` of course then breaks because it uses most current (2.2).  Using `specVersion="2.1"` still seems to run with XJC 2.2 even though source may be 2.1 compatible so error still occurs.

Using `maven-jaxb21-maven` works perfectly except when I switch to it, m2e no longer adds `target/generated-sources/xjc` as a source dir, etc.  I tried looking to see if I could see why but am not a maven plugin expert.  I also tested `maven-jaxb22-maven` and it behaved the same.

TLDR; `maven-jaxb2-plugin` works with m2e perfectly.  The JAXB specific versions `maven-jaxb2x-plugin` work as described but lose control over m2e.

**Comment by DeezCashews** [comment]
Changed `maven-jaxb21-plugin` version from `0.11.0` to `0.13.1` and issue went away.  Didn't realize until it hit me why I couldn't find culprit in source and then saw version of source.

**Comment by highsource** [comment]
I just feel I have to keep silent and all the issues will go away. :)

**Comment by DeezCashews** [comment]
As a developer it pains me when I sound like a BA ;)  Out of curiosity, what part of the plugin ends up modifying Eclipses .classpath and .settings/org.eclipse.wst.common.component?  I'm guessing m2e does it but how do you "natively" tell m2e what to add?  Is it as simple as modifying the `MavenProject` and m2e picks it up?

**Comment by highsource** [comment]
There's nothing `m2e` specific, I just [add "compile source root" to Maven model](https://github.com/highsource/maven-jaxb2-plugin/blob/667f8caf18b5d6f8ca8cdfe9ac1351c14b96f37e/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/RawXJC2Mojo.java#L616). So from the model POV this this normal Maven functionality. `m2e` just picks up what the Maven model says.
